src/driver/transport/ssh.cr

```diff
@@ -81,13 +81,39 @@ class PlaceOS::Driver
           # Attempt to authenticate
           supported_methods = session.login_with_noauth(settings.username)
           if supported_methods
-            if password = settings.password
-              session.login(settings.username, password)
-            end
-
-            if prikey = settings.private_key
-              pubkey = settings.public_key.not_nil!
-              session.login_with_data(settings.username, prikey, pubkey, settings.passphrase.try &.to_slice.to_unsafe)
+            if supported_methods.is_a?(Array(String))
```

Comment:
The `supported_methods` returned from libssh2 is a comma separated string. This should check `.is_a?(String)`, then split if so.

Reply:
I've already updated the shard to return an array of strings